sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -325,11 +325,12 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       val dsOptions = new CaseInsensitiveStringMap(finalOptions.asJava)
 
       def getTable: Table = {
-        // For file source, it's expensive to infer schema/partition at each write. Here we pass
-        // the schema of input query and the user-specified partitioning to `getTable`. If the
+        // If the source accepts external table metadata, here we pass the schema of input query
+        // and the user-specified partitioning to `getTable`. This is for avoiding
+        // schema/partitioning inference, which can be very expensive. If the
         // query schema is not compatible with the existing data, the write can still success but
```

Comment:
This is only true for file source. How about
```
If the query schema is not compatible with the existing data, the behavior is undefined.
For example, file source write will success but following reads would fail.
```

Reply:
+1, thanks